src/main/java/org/sigmah/client/util/profiler/Profiler.java

```diff
@@ -203,110 +139,64 @@ public void onSuccess(Request result) {
-				private final DateTimeFormat formatter = DateTimeFormat.getFormat(DateTimeFormat.PredefinedFormat.DATE_TIME_FULL);
+			private final StringBuilder csvBuilder = new StringBuilder("scenario;duration;date;version;user-agent\n");
+			private final DateTimeFormat formatter = DateTimeFormat.getFormat(DateTimeFormat.PredefinedFormat.DATE_TIME_FULL);
 
-				@Override
-				public void onSuccess(Request result) {
-					final Cursor cursor = result.getResult();
-					if(cursor != null) {
-						final Execution execution = cursor.getValue();
+			@Override
+			public void onFailure(Throwable caught) {
+				callback.onFailure(caught);
+			}
+
+			@Override
+			public void onSuccess(Execution execution) {
+				if (execution != null) {
+					if (execution.isOnline() && ApplicationCache.Status.IDLE.name().equals(execution.getApplicationCacheStatus())) {
 						csvBuilder.append(execution.getScenarioName()).append(';')
-								.append(execution.getDuration()).append(';')
+								.append(formatDoubleForCSV(execution.getDuration())).append(';')
 								.append(formatter.format(execution.getDate())).append(';')
 								.append(execution.getVersionNumber()).append(";\"")
-								.append(execution.getUserAgent()).append("\"\n");
-						
-						cursor.next();
-					} else {
-						callback.onSuccess(csvBuilder.toString());
+								.append(execution.getUserAgent()).append('\"');
+
+						for (final Checkpoint checkpoint : execution.getCheckpointSequence()) {
+							csvBuilder.append(';').append(checkpoint.getName())
+									.append(';').append(formatDoubleForCSV(checkpoint.getDuration()));
+						}
+
+						csvBuilder.append("\r\n");
 					}
+				} else {
+					callback.onSuccess(csvBuilder.toString());
 				}
-			});
-		} else {
-			callback.onSuccess("IndexedDB database is not opened.");
-		}
-	}
-
-	/**
-	 * Generates an identifier for the given execution.
-	 * 
-	 * @param execution Execution to use.
-	 * @return An identifier.
-	 */
-	private String identifierForExecution(final Execution execution) {
-		final Date date = execution.getDate();
-		if (date != null) {
-			return execution.getScenarioName() + '-' + date.getTime();
-		} else {
-			return execution.getScenarioName() + "-PAUSE";
-		}
+			}
+		});
 	}
 	
-	/**
-	 * Calculates the duration of the given execution.
-	 * 
-	 * @param execution Execution to measure.
-	 * @return Duration of the given execution.
-	 */
-	private double durationOfExecution(final Execution execution) {
-		double duration = execution.getDuration();
-		
-		final Date date = execution.getDate();
-		if (date != null) {
-			duration += (new Date().getTime() - date.getTime()) / 1000.0;
-		}
+	public void generateMarkdownReport() {
 		
-		return duration;
 	}
 	
 	/**
-	 * Format the current duration.
+	 * Format the given double value to a String suitable for a CSV file.
 	 * 
-	 * @param duration Duration to format.
-	 * @return The duration formatted.
+	 * @param value Value to format.
+	 * @return The formatted value.
```

Comment:
`@return The formatted value` is pretty obvious, can you elaborate? Actually, just add sample output should be sufficient (id `formatDuration(123456789) will return ...`)